Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/EditorComponentBase.cpp

```diff
@@ -102,9 +106,21 @@ namespace AzToolsFramework
             m_alias = alias;
         }
 
-        AZStd::string EditorComponentBase::GetSerializedIdentifier() const
+        AZStd::string EditorComponentBase::GetSerializedIdentifier()
         {
+            // Initialize if necessary. This is needed because not all derived components are calling their base class Init()
+            if (m_alias.empty())
+            {
+                InitSerializedIdentifier();
```

Comment:
Instead of leaving letting alias be created in any Get() call, I think it would be easier to maintain this if we explicitly create the alias in EditorComponentBase() constructor. If the id is empty by the time the constructor is called, you can create the id there too doing this : https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Component/Component.cpp#L156

The id gets set later at some point when SetEntity() is called on the component. So, I think its fine to call this during the component creation itself, (particularly because the creation of id seems to have nothing to do with entity data itself).

Reply:
The Id is always going to be empty when the constructor.
It is the constructor that initializes the data.

The steps of initialization that are relevant to determining the component ID are as follows
1. Component construction - Component ID is not set yet.
1. (If Component is loaded from serialized data) Entity de-serialization - Component ID can be loaded from prefab data
1. Entity::Init - Calls `Component::SetEntity`  to set Component ID

Reply:
I'm not sure if it makes sense to initialize the alias in the constructor because a component that is loaded from serialized data will set that value to something else.

Reply:
I think the fact that deserialization may alter the alias is fine imo. The problem this PR is trying to solve is , "I created a new component in the editor, I don't know what its alias should be". For cases where we load the component from prefab data, there is currently no issue. We simply read it from json.

However, if we create a new component, we don't want to deserialize again immediately just to give it a unique alias. So, starting with a placeholder alias in the constructor sounds like a reasonable approach to me. If the id is empty in the constructor, we can create that too. I can't think of that creating any issues.

Reply:
That breaks an invariant that the Component Id is set to `InvalidComponentId` if it is not attached to an entity.

There is code that uses that constraint to determine how to lookup a component on an entity (https://github.com/o3de/o3de/blob/9ef05ce7fa90d6299e384af775c27ca680cd5f7a/Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EntityUtilityComponent.cpp#L124-L131).

Reply:
We don't have to incorporate componentId to construct the alias. We can generate our own Id without referencing componentId.

Reply:
Because the ComponentId is set, it wouldn't go into the else block that uses the `AZ::TypeId`, instead it would use the created component Id which should work in all circumstances unless their is a random number collision which should be on the order of 2^64.